src/template.js

```diff
@@ -1843,43 +1843,69 @@ module.exports = {
-                        }
-                    },
-                    'q-applicant-last-name': {
-                        title: 'Last name',
-                        type: 'string',
-                        maxLength: 70,
+                allOf: [
+                    {
+                        title: 'Enter your name',
+                        required: [
+                            'q-applicant-title',
+                            'q-applicant-first-name',
+                            'q-applicant-last-name'
+                        ],
+                        propertyNames: {
+                            enum: [
+                                'q-applicant-title',
+                                'q-applicant-first-name',
+                                'q-applicant-last-name'
+                            ]
+                        },
+                        allOf: [
+                            {
+                                properties: {
+                                    'q-applicant-title': {
+                                        title: 'Title',
+                                        type: 'string',
+                                        maxLength: 6,
+                                        errorMessage: {
+                                            maxLength: 'Title must be 6 characters or less'
+                                        }
+                                    }
+                                }
+                            },
+                            {
+                                properties: {
+                                    'q-applicant-first-name': {
+                                        title: 'First name',
+                                        type: 'string',
+                                        maxLength: 70,
+                                        errorMessage: {
+                                            maxLength: 'First name must be 70 characters or less'
+                                        }
+                                    }
+                                }
+                            },
+                            {
+                                properties: {
+                                    'q-applicant-last-name': {
+                                        title: 'Last name',
+                                        type: 'string',
+                                        maxLength: 70,
+                                        errorMessage: {
+                                            maxLength: 'Last name must be 70 characters or less'
+                                        }
+                                    }
+                                }
+                            }
+                        ],
                         errorMessage: {
-                            maxLength: 'Last name must be 70 characters or less'
+                            required: {
```

Comment:
have you tested both the `maxLength` an the `required` error messages are working for this new shape?

Reply:
I've "technically" not changed anything. I moved the body of the existing schema into an `allOf`. It's still validating the same shape of data and the valid/invalid examples are all passing as before. The error stuff works off of the data paths, not the schema paths, so the bits of the errors we use haven't changed.

In regards to the shape of the errors... It's maybe something we should add tests for going forward. I'm pretty sure the newer JSON Schema specs have standardised the error format, so should be pretty easy. Only tricky bit is, we have this template as a dependency at the moment, so running the tests *here* doesn't make much sense 🤔